index.js

```diff
@@ -379,3 +379,112 @@ exports.requestMultiItemBasic = (itemIds, cb) => {
+    console.error('No credentials set')
+    if (cb) cb('No credentials set', false)
+  } else {
+    // uses the basic auth method to ask for the token
+    return new Promise((resolve, reject) => {
+      request.post(exports.credsBase + 'token', {
+        'auth': {
+          'user': exports.credsKey,
+          'pass': exports.credsSecret
+        }
+      },
+      (error, response, body) => {
+        if (error) console.error(error)
+        if (!response) {
+          console.error('Error: Make sure you set the correct base path to the API.')
+          exports.authorizedTimestamp = false
+          if (cb) cb('Error: Make sure you set the correct base path to the API.', false)
+        }
+
+        if (response.statusCode === 200) {
+          exports.authorizedToken = JSON.parse(body)['access_token']
+          exports.authorizedTimestamp = Math.floor(Date.now() / 1000)
+
+          // if that worked we want to keep the token always up to date, so request a new one every halflife
+          // clearInterval(exports.authorizedTimer)
+          // exports.authorizedTimer = setInterval(() => {
+          //   exports.auth(() => {
+          //     // do something on the refresh?
+          //   })
+          // }, ((JSON.parse(body)['expires_in'] / 2) * 1000))
+
+          if (cb) {
+            resolve(cb(null, JSON.parse(body)['access_token']))
+          }
+        } else {
+          console.error('error: ' + response.statusCode)
+          exports.authorizedTimestamp = false
+          if (cb) cb(body, false)
+        }
+      })
+    })
+  }
+}
+
+exports.apiGet = (path, cb) => {
+  if (!exports.authorizedToken) {
+    console.error('No authorizedToken set')
+    if (cb) cb('No authorizedToken set', false)
+  } else {
+    var url = `${exports.credsBase}${path}`
+    console.log(url)
+    // use the bearer auth token
+    return new Promise((resolve, reject) => {
+      request.get(url, {
+        'timeout': 120 * 1000,
+        'auth': {
+          'bearer': exports.authorizedToken
+        }
+      },
+      (error, response, body) => {
+        if (error) console.error(error)
+        if (response.statusCode && response.statusCode === 200) {
+          var result = {}
+          result.total = 1
+          result.entries = []
+          result.entries[0] = JSON.parse(body)
+          if (cb) {
+            resolve(cb(null, {data: result, url: url}))
+          }
+        } else {
+          if (cb) reject(cb(body, false))
+        }
+      })
+    })
+  }
+}
+
+exports.apiPost = (path, data, cb) => {
+  if (!exports.authorizedToken) {
+    console.error('No authorizedToken set')
+    if (cb) cb('No authorizedToken set', false)
+  } else {
+    var url = `${exports.credsBase}${path}`
+    console.log(url)
+    // use the bearer auth token
+    return new Promise((resolve, reject) => {
+      request(Object.assign(data, { 'url': url, 'auth': {'bearer': exports.authorizedToken} }),
+      (error, response, body) => {
```

Comment:
Same here, it'd be good to indent everything under `request` from 476 to 487.

Reply:
Done